sdks/java/core/src/main/java/org/apache/beam/sdk/io/OffsetBasedSource.java

```diff
@@ -337,7 +337,7 @@ public boolean allowsDynamicSplitting() {
     }
 
     @Override
-    public final synchronized OffsetBasedSource<T> splitAtFraction(double fraction) {
+    public synchronized OffsetBasedSource<T> splitAtFraction(double fraction) {
```

Comment:
I do not know why a source that works purely on an offset range would want to override this behavior. This is final intentionally since splitting for all offset-based sources is expected to be this way. If splitting behavior of BQ Storage API source is significantly different I think it might be better to just copy and adapt this code rather than overriding OffsetBasedSource.